internal/testutil/mig/provider.go

```diff
@@ -32,11 +33,19 @@ func ExternalProvidersWithAWS() map[string]resource.ExternalProvider {
 
 func checkLastVersion(tb testing.TB) {
 	tb.Helper()
+	validateConflictingEnvVars()
 	if os.Getenv("MONGODB_ATLAS_LAST_VERSION") == "" {
 		tb.Fatal("`MONGODB_ATLAS_LAST_VERSION` must be set for migration acceptance testing")
 	}
 }
 
 func versionConstraint() string {
+	validateConflictingEnvVars()
 	return os.Getenv("MONGODB_ATLAS_LAST_VERSION")
 }
+
+func validateConflictingEnvVars() {
+	if os.Getenv("TF_CLI_CONFIG_FILE") != "" {
+		log.Fatal("found `TF_CLI_CONFIG_FILE` in env-var when running migration tests, this might override the terraform provider for MONGODB_ATLAS_LAST_VERSION and instead use local latest build!")
```

Comment:
Would have liked to use `tb.Fatal` instead of `log.Fatal` but I think there is a possibility of `versionConstraint` being called without `checkLastVersion`, therefore, I had to use `log.Fatal`

Reply:
sorry, not sure I understand this, can you elaborate or explain in a different way?  We shouldn't use log.Fatal in tests but tb.Fatal

Reply:
I was afraid of a case where some migration tests would call `ExternalProvidersWithAWS` and `ExternalProviders` which call `versionConstraint` and the migration test would not be skipped.
However, looking at the code, everywhere those are used we expect to see `mig.PreCheck` or `mig.SkipIfVersionBelow` used which would check the TF_CLI_CONFIG_FILE env-var and skip the test